services-api/src/main/java/io/scalecube/services/RemoteService.java

```diff
@@ -0,0 +1,4 @@
+package io.scalecube.services;
+
+/** Interface-marker for Proxy of remote services. */
+public interface RemoteService {}
```

Comment:
Seems like workaround with bad name.
And I see you use it in factories like the following:

```java
    @Override
    public Mono<? extends Collection<ServiceInfo>> initializeServices(
        MicroservicesContext microservices) {
      return Mono.fromCallable(
          () -> {
            this.context.start();
            return this.context.getBeansWithAnnotation(Service.class).values().stream()
                .filter(
                    Predicate.not(bean -> RemoteService.class.isAssignableFrom(bean.getClass())))
                .map(bean -> ServiceInfo.fromServiceInstance(bean).build())
                .collect(Collectors.toList());
          });
    }
```

Could we replace 

```java
RemoteService.class.isAssignableFrom(bean.getClass())
```

to use smth like?: 

```java
Reflect.isService(bean.getClass())
```

Reply:
We need way for detecting proxy for remote service and local service. 
`Reflect.isService(bean.getClass)` don't do it

Reply:
Assume, but you have in arguments `MicroservicesContext` instance and you can filter out something via its providing API.

Reply:
MicroservicesContext cant resolve this problem. MicroservicesContext need for inject this in other services, no more.

Reply:
please, see picture in top this PR

Reply:
I'd like to remove this class.
Look the given context has `io.scalecube.services.MicroservicesContext#serviceDiscovery` and this one has `io.scalecube.services.discovery.api.ServiceDiscovery#serviceEndpoint` so you have the ability to understand what this instance provides (local call) and what not (remote call).

Reply:
Or let's just provide a new method which will give us them in `MicroservicesContext`.
like `Collection<Class> services()`

Reply:
Yep, ok, this is redundant. But, how do detect proxy and local services in guice? I don't know :)

Reply:
Please reread the above my comments

Reply:
fixed